nxcompshad/src/Logger.h

```diff
@@ -29,6 +29,9 @@
 #include <cerrno>
 #include <cstdarg>
 
+#pragma GCC diagnostic push
```

Comment:
This will only work for GCC... and maybe fail on older versions. I'm not sure since when `-Wformat-nonliteral` exists. Looks like it has been around since at least 3.4, so that's good enough.

We may want this for clang, too?

Let's go for

```C
#ifdef __clang__
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wformat-nonliteral"
#else
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wformat-nonliteral"
#endif
```
and

```C
#ifdef __clang__
#pragma clang diagnostic pop
#else
#pragma GCC diagnostic pop
#endif
```

Reply:
according to the clang documentation clang also understands gcc pragmas(https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-via-pragmas):
 
"Clang can also control what diagnostics are enabled through the use of pragmas in the source code. This is useful for turning off specific warnings in a section of source code. Clang supports GCC’s pragma for compatibility with existing source code, as well as several extensions."

See also https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-via-pragmas